src/test/java/pro/trevor/tankgame/Rule/BuyActionWithGoldTest.java

```diff
@@ -0,0 +1,166 @@
+package pro.trevor.tankgame.Rule;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import pro.trevor.tankgame.rule.definition.player.PlayerActionRule;
+import pro.trevor.tankgame.rule.impl.shared.PlayerRules;
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.rule.impl.version3.TankAttribute;
+import pro.trevor.tankgame.state.State;
+
+public class BuyActionWithGoldTest 
+{
+    @Test
+    public void TypeCheckTest()
+    {
+        PlayerActionRule<Tank> rule = PlayerRules.BuyActionWithGold(3); // action_cost is a don't care
+        assertEquals(PlayerRules.ActionKeys.BUY_ACTION, rule.name(), "Asserts that the buy action with gold rule is the right type");
+    }
+
+    private Tank BuildTestTank(int actions, int gold, boolean dead)
+    {
+        JSONObject json = new JSONObject();
```

Comment:
This is very verbose and probably something that a lot of tests are going to have to do.  There should probably be a helper function for creating generic tanks?  Trevor will have an opinion.

Reply:
Moved to TestUtilities